substrate/frame/balances/src/tests/mod.rs

```diff
@@ -278,6 +279,23 @@ pub fn info_from_weight(w: Weight) -> DispatchInfo {
 	DispatchInfo { weight: w, ..Default::default() }
 }
 
+/// Check that the total-issuance matches the sum of all accounts' total balances.
+pub fn ensure_ti_valid() {
```

Comment:
Can we easily inject this at the end of all tests in this crate?

Reply:
Ah yes. I though about adding a `fn register_cleanup_hook` to the `TestExternalities`.  
Then we could set this in the externalities as hook and have it run at the end without any refactoring. Otherwise i think we need to refactor to not directly use the `execute_with` but wrap it again.

Reply:
I would be curious to also know if it's feasible as a try-state hook

Reply:
> Otherwise i think we need to refactor to not directly use the execute_with but wrap it again.

Yeah this is what most pallets do, and I generally find it easier. 

> I would be curious to also know if it's feasible as a try-state hook

Ideally it should be, but yeah iterating all accounts will ruin everything else 🙈 We need to think of a system to separate try-state hooks that we always run in a place like CI vs. those that we want to run every month etc.

Reply:
I would be really interested to bench it. We have a lot of really heavy staking hooks today without much issue, maybe as long as it's O(n) it's OK.

Reply:
looking forward to adding it :)

Reply:
I ran these checks in remote-externalities to fine some issues with this: https://github.com/ggwpez/wtfwt  
takes like 3-5 secs for one run on Polkadot.

Reply:
3-5s is acceptable imo